### PR TITLE
ENH: enable out-of-place-registration

### PR DESCRIPTION
**Pull request by ambrosejcarr**
Enables out-of-place registration.

### CODE REVIEW

**Comment by ambrosejcarr** [comment]
Rebased and confirmed this works on master with Tony's cache fix.

File: starfish/image/_registration/fourier_shift.py [hunk 1]
```diff
@@ -34,7 +48,25 @@ def add_arguments(cls, group_parser) -> None:
...
+        Parameters
+        ----------
+        image : ImageStack
+            The stack to be registered
+        in_place : bool
+            If false, return a new registered stack. Else, register in-place (default False)
+
+        Returns
+        -------
+
```
Comment (ttung) [inline starfish/image/_registration/fourier_shift.py]: stuff missing.

File: starfish/image/_registration/fourier_shift.py [hunk 2]
```diff
@@ -54,7 +86,9 @@ def run(self, image: ImageStack) -> ImageStack:
                     result = shift_im(data, shift)
                     image.set_slice(indices=indices, data=result)
 
-        return image
+        if not in_place:
+            return image
+        return None
```
Comment (ttung) [inline starfish/image/_registration/fourier_shift.py]: technically not required.

Comment (ambrosejcarr) [reply starfish/image/_registration/fourier_shift.py]: Mypy got upset without it there at one point. I will check if it gets upset here, and if not will remove it.

**Comment by ttung** [comment]
> Generally speaking, I think we should have image->image pipeline components return ImageStack, even if it's in-place. You can string up a bunch of operations together that way.

curious what you think of ^^, in case you missed it. :)

**Comment by ambrosejcarr** [comment]
> Generally speaking, I think we should have image->image pipeline components return ImageStack, even if it's in-place. You can string up a bunch of operations together that way.

Yep, I see the advantage and am fine with that.

**Pull request merged**